util/conditions/v1beta2/summary.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// SummaryOption is some configuration that modifies options for a summary call.
+type SummaryOption interface {
+	// ApplyToSummary applies this configuration to the given summary options.
+	ApplyToSummary(*SummaryOptions)
+}
+
+// SummaryOptions allows to set options for the summary operation.
+type SummaryOptions struct {
+	mergeStrategy                  MergeStrategy
+	conditionTypes                 []string
```

Comment:
What happens if the conditionTypes option is not set (or empty)? Is this a supported case? If not should this become a "mandatory" parameter of the NewSummaryCondition func? (including input validation that it cannot be empty)

Reply:
Discussed, going to keep the option, we'll just validate it as "input validation" at the beginning of the func

Reply:
done